project.clj

```diff
@@ -129,17 +138,36 @@
                                                      district0x.ui.history.pushroute-hosts "localhost"
                                                      name-bazaar.ui.db.log-level "debug"}
                                    :external-config {:devtools/config {:features-to-install :all}}}}
+
+                       ;; Development on client-side UI, with mainnet
+                       {:id "dev-ui-only"
+                        :source-paths ["src"]
```

Comment:
If we want to exclude them from each other, then they need to be separated differently, since the `:source-path` is supposed to resemble the classpath when performing namespace resolution. At least this is normal behaviour for clojure namespaces, so maybe the cljsbuild `:source-path` is more lenient. In other words, I don't know how the original implementation even works, because it technically shouldn't if it was following the rules.

This could pose a problem if we start using build tools that don't make use of cljsbuild, or if they choose to start doing namespace resolution the same way clojure does namespace resolution through the directory structure.

Isolating the code is a good idea, the directory structure needs to change to reflect the namespace.

I'll try and explain the problem more clearly:

before for "dev" build id of cljsbuild, it would resolve the sourcepaths
- "src/name_bazaar/ui"
- "src/name_bazaar/shared"
- "src/district0x/ui"
- "src/district0x/shared"

As an example, looking in the file `src/name_bazaar/ui/core.cljs`, we have the `(require)`'d namespace:

`[district0x.ui.events]`
;;...


Typically in clojure, the namespace resolution will look for a *.class, *.cljc, *.clj file in:
- {source-path}/district0x/ui/events.class
- {source-path}/district0x/ui/events.clj
- {source-path}/district0x/ui/events.cljc

similarly for clojurescript:

- {source-path}/district0x/ui/events.cljs

but look at our source-path, none of the source-paths reflect this, so cljsbuild is doing namespace resolution in an undocumented way.

The way to fix it would be to restructure the directories as such:

- src/server/name_bazaar/server
- src/server/district0x/server

- src/shared/name_bazaar/shared
- src/shared/district0x/shared

- src/ui/name_bazaar/ui
- src/ui/district0x/ui

The new `:source-path` for "dev-ui" would be ["src/shared" "src/ui"]

The new `:source-path` for "dev-server" would be ["src/shared" "src/server"]